### PR TITLE
server, tests: remove EnableZap

### LINKED ISSUE #5118: Remove `EnableZap` since it's no longer needed
## Enhancement Task

`EnableZap` is used to set the etcd logger in tests.

https://github.com/tikv/pd/blob/147e9c041a49155e3c15ec0432cf9de1d4d38c44/server/server.go#L85-L87

However, this `cfg.Logger` has been set by default here. So it's unnecessary, and could be removed.

https://github.com/tikv/pd/blob/147e9c041a49155e3c15ec0432cf9de1d4d38c44/server/config/config.go#L1297

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5118.

### What is changed and how does it work?

```commit-message
Remove `EnableZap`.
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -282,14 +278,6 @@ func CreateServer(ctx context.Context, cfg *config.Config, serviceBuilders ...Ha
 		diagnosticspb.RegisterDiagnosticsServer(gs, s)
 	}
 	s.etcdCfg = etcdCfg
-	if EnableZap {
-		// The etcd master version has removed embed.Config.SetupLogging.
-		// Now logger is set up automatically based on embed.Config.Logger,
-		// Use zap logger in the test, otherwise will panic.
-		// Reference: https://go.etcd.io/etcd/blob/master/embed/config_logging.go#L45
-		s.etcdCfg.Logger = "zap"
-		s.etcdCfg.LogOutputs = []string{"stdout"}
```
Comment (rleungx) [inline server/server.go:291]: Should it be added into `GenEmbedEtcdConfig`?

Comment (JmPotato) [reply server/server.go:291]: By default, this will be set to "stderr". Is there any strong need to change it to "stdout" in the test?

**Review by rleungx — APPROVED** [review]

**Review by AndreMouche — APPROVED** [review]
LGTM

File: server/server.go (hunk 1 above)
Comment (nolouch) [reply server/server.go:291]: can we see the etcd relative log in the log file?

**Pull request merged**